src/project.cc

```diff
@@ -572,13 +574,20 @@ std::vector<Project::Entry> LoadCompilationEntriesFromDirectory(
     // Try to load compile_commands.json, but fallback to a project listing.
   } else {
     project->mode = ProjectMode::ExternalCommand;
-#if defined(_WIN32)
-// TODO
-#else
-    char tmpdir[] = "/tmp/cquery-compdb-XXXXXX";
-    if (!mkdtemp(tmpdir))
-      return {};
-    comp_db_dir = tmpdir;
+
+    // generally it would be nice if we could just let clang load the
+    // compilation database (compile_commands.json) from memory.
+    // However, clang insists on reading compile_commands.json from a
+    // directory, so we create a temporary directory just for clang to read
+    // from.
+
+    char templ[] = "/tmp/cquery-compdb-XXXXXX";
+    auto tmpdir = TryMakeTempDirectory(templ);
+    if(!tmpdir.has_value()) {
+        return {};
```

Comment:
return nullopt

Reply:
Can't return `nullopt`, must be a `std::vector`.